stonesoup/types/particle.py

```
@@ -25,3 +30,58 @@ def __init__(self, state_vector, weight, parent=None, *args, **kwargs):
     @property
     def ndim(self):
         return self.state_vector.shape[0]
+
+
+class Particles(Type):
+    """
+    Particle type
+
+    A collection of particles. Contains a state and weight for each particle
+    """
+    state_vector: StateVectors = Property(default=None, doc="State vectors of particles")
+    weight: MutableSequence[Probability] = Property(default=None, doc='Weights of particles')
+    parent: 'Particles' = Property(default=None, doc='Parent particles')
+    particle_list: MutableSequence[Particle] = Property(default=None,
+                                                        doc='List of Particle objects')
+
+    def __init__(self, state_vector=None, weight=None, parent=None, particle_list=None,
+                 *args, **kwargs):
+        if state_vector is not None:
+            assert particle_list is None,\
```

This should probably raise an error instead of attempting to assert first.